variables.tf

```diff
@@ -1,42 +1,149 @@
 variable "region" {
   description = "The region used to launch this module resources."
+  type        = string
   default     = ""
 }
 
+variable "profile" {
+  description = "The profile name as set in the shared credentials file. If not set, it will be sourced from the ALICLOUD_PROFILE environment variable."
+  type        = string
+  default     = ""
+}
+variable "shared_credentials_file" {
+  description = "This is the path to the shared credentials file. If this is not set and a profile is specified, $HOME/.aliyun/config.json will be used."
+  type        = string
+  default     = ""
+}
+
+variable "skip_region_validation" {
+  description = "Skip static validation of region ID. Used by users of alternative AlibabaCloud-like APIs or users w/ access to regions that are not public (yet)."
+  type        = bool
+  default     = false
+}
+
 # VSwitch variables
 variable "vswitch_id" {
   description = "The vswitch id used to launch load balancer."
+  type        = string
   default     = ""
 }
 
 # Load Balancer Instance variables
+
+variable "use_existing_slb" {
+  description = "Whether to use an existing load balancer instance. If false, you can create a new one by set 'create' to true."
```

Comment:
"Whether to use an existing load balancer instance. If true, 'existing_slb_id' should not be empty. Also, you can create a new one by setting 'create =  true'."